### PR TITLE
Adding user aggregates on change

### PR DESCRIPTION
**Pull request by manasnisar**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | anas/fix/client-aggrs  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by dabasov — APPROVED** [review]
lgtm

File: code/go/0chain.net/smartcontract/dbs/event/user_aggregate.go [hunk 1]
```diff
@@ -19,99 +19,244 @@ type UserAggregate struct {
...
+	},
+}
 
-	var currentUsers []User
-	result := edb.Store.Get().Model(&User{}).
-		Where("users.user_id in (select id from temp_user_ids ORDER BY id limit ? offset ?)", limit, offset).
-		Find(&currentUsers)
+func (edb *EventDb) GetLatestUserAggregates(ids map[string]interface{}) (map[string]*UserAggregate, error) {
+	var ua []UserAggregate
+	var mappedAggrs = make(map[string]*UserAggregate, len(ua))
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/dbs/event/user_aggregate.go]: Should be `len(ids)`

File: code/go/0chain.net/smartcontract/dbs/event/user_aggregate.go [hunk 2]
```diff
@@ -19,99 +19,244 @@ type UserAggregate struct {
...
+		if h, ok := handlers[ev.Tag]; ok {
+			aggrs := h(ev)
+			updatedAggrs = append(updatedAggrs, aggrs...)
 		}
 	}
 
-	oldUsers, err := edb.getUserSnapshots(limit, offset)
+	ids := make(map[string]interface{})
+	for _, aggr := range updatedAggrs {
+		aggr.Round = e.round
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/dbs/event/user_aggregate.go]: `aggr` is value, not pointer, `aggr.Round = e.round` would not change it actually.

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Review by peterlimg — APPROVED** [review]
LGTM

**Pull request merged**